monitor/services/src/test/java/org/datacleaner/monitor/server/controllers/ReferenceDataControllerTest.java

```diff
@@ -0,0 +1,217 @@
+                new DictionaryModel("dictionary", Arrays.asList("foo", "bar", "baz"), true);
+
+        final MvcResult mvcResult = _mockMvc.perform(
+                put("/test/referencedata/dictionary/testDictionary").contentType(MediaType.APPLICATION_JSON)
+                        .content(_objectMapper.writeValueAsString(dictionaryModel))).andExpect(status().isCreated())
+                .andExpect(header().string("Location",
+                        is("http://localhost/test/referencedata/dictionary/testDictionary"))).andReturn();
+
+        final String location = mvcResult.getResponse().getHeader("Location");
+        _mockMvc.perform(get(location)).andExpect(status().isOk()).andExpect(jsonPath("$.name", is("testDictionary")))
+                .andExpect(jsonPath("$.entries", hasSize(3))).andExpect(jsonPath("$.entries[2]", is("baz")))
+                .andExpect(jsonPath("$.caseSensitive", is(true)));
+        _mockMvc.perform(delete(location)).andExpect(status().isNoContent());
+        _mockMvc.perform(get(location)).andExpect(status().isNotFound());
+    }
+
+    @Test
+    public void testPutDictionaryFile() throws Exception {
+        final byte[] bytes = Files.readAllBytes(Paths.get("src/test/resources/testdictionary"));
+        final MvcResult mvcResult = _mockMvc.perform(
+                put("/test/referencedata/dictionary/testDictionary").content(bytes).param("casesensitive", "false"))
+                .andExpect(status().isCreated()).andExpect(header().string("Location",
+                        is("http://localhost/test/referencedata/dictionary/testDictionary"))).andReturn();
+
+        final String location = mvcResult.getResponse().getHeader("Location");
+        _mockMvc.perform(get(location)).andExpect(status().isOk()).andExpect(jsonPath("$.name", is("testDictionary")))
+                .andExpect(jsonPath("$.entries", hasSize(3))).andExpect(jsonPath("$.entries[2]", is("baz")))
+                .andExpect(jsonPath("$.caseSensitive", is(false)));
+    }
+
+    private RequestPostProcessor withPut() {
+        return request -> {
+            request.setMethod("PUT");
+            return request;
+        };
+    }
+
+    @Test
+    public void testListSynonymCatalogs() throws Exception {
+        _mockMvc.perform(get("/test/referencedata/synonymCatalogs")).andExpect(status().isOk())
+                .andExpect(content().contentType(MediaType.APPLICATION_JSON)).andExpect(jsonPath("$", hasSize(1)))
+                .andExpect(jsonPath("$[?(@.referenceDataType!='SYNONYM_CATALOG')]", hasSize(0)))
+                .andExpect(jsonPath("$[?(@.referenceDataType=='SYNONYM_CATALOG')]", hasSize(1))).andExpect(
+                jsonPath("$[0].href", is("http://localhost/test/referencedata/synonymCatalog/job%20title%20synonyms")));
+    }
+
+    @Test
+    public void testPutAndDeleteSynonymCatalog() throws Exception {
+        final SynonymCatalogModel dictionaryModel = new SynonymCatalogModel("synonymCatalog",
```

Comment:
The variable is called `dictionaryModel`, but it contains a `SynonymCatalogModel` so preferably rename it to `synonymCatalogModel`.